app/src/main/java/com/odysee/app/tasks/CommentOptionTask.java

```diff
@@ -0,0 +1,109 @@
+package com.odysee.app.tasks;
+
+import android.os.AsyncTask;
+
+import com.odysee.app.adapter.CommentOption;
+import com.odysee.app.exceptions.ApiCallException;
+import com.odysee.app.model.Comment;
+import com.odysee.app.utils.Comments;
+import com.odysee.app.utils.Helper;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+/**
+ * Performs the network call for a {@link CommentOption}
+ */
+public class CommentOptionTask extends AsyncTask<Void, Void, Void> {
```

Comment:
DON’T use AsyncTask as they are deprecated and using threads along Futures or Supplier will work without the need of using artificially added classes like AsyncTask

Reply:
Agreed! I was copying similar code I saw that uses AsyncTask, erring on the side of consistency with existing code in the project. Perhaps I will come up with a generic replacement and replace AsyncTask in a few other places when I get back to this PR.